### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by chrisbolin**
- react tests
- console.warn tests
- update eslint for es6 and mocha

### CODE REVIEW

File: spec/react.spec.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+class Container extends React.Component {
+  render() {
+    return React.createElement(ChildWithShouldComponentUpdate, {
+      children: React.createElement('div')
+    });
+  }
+}
+
+describe('advanced', () => {
+  let warnStub = sinon.stub(console, 'warn');
```
Comment (ryan-roemer) [inline spec/react.spec.js]: GLOBAL: Never _set_ something in a `describe`. Leave it open like:

```js
let warnStub;
```

Then set in `beforeEach`.

Comment (ryan-roemer) [reply spec/react.spec.js]: GLOBAL: Technically optional, but **strongly** recommended. Use a `sandbox` instead. Sandboxes track fakes and globally restore saving you the pain of having to manually restore. So, here putting together all the comments:

```js
describe('advanced', () => {
  let sandbox;
  let warnStub;

  beforeEach(() => {
    sandbox = sinon.sandbox.create();
    warnStub = sandbox.stub(console, "warn");
  });

  afterEach(() => {
    sandbox.restore();
  });
```

You can move that sandbox creation to a higher level `beforeEach` or even a base-but-always-included global `beforeEach|afterEach` too if you need fakes across a test suite -- reusing the _same_ sandbox is a good practice and I can help fill in the logistical details for approaches for this if you need this in the future 😉

File: spec/react.spec.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        const testRenderer = ReactTestRenderer.create(React.createElement(Container));
+        testRenderer.update(React.createElement(Container));
+        assert.strictEqual(warnStub.notCalled, true);
+      });
+    });
+  });
+
+  describe('warnings', () => {
+    describe('generic circular references', () => {
+      it('generate a warning but do not throw an error', () => {
```
Comment (ryan-roemer) [inline spec/react.spec.js]: NIT: `generates`

File: package.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "eslint": "eslint *.js benchmark spec",
+    "test-spec-watch": "mocha spec/*.spec.js -R spec --watch",
```
Comment (ryan-roemer) [inline package.json]: You don't really need this. Simply run:

```sh
$ yarn run test-spec --watch
```

and you can just inject the "extra" flags... If you can't remember `--watch`, then it's cool to keep this.

File: spec/react.spec.js (hunk 1 above)
Comment (chrisbolin) [reply spec/react.spec.js]: sounds great!

File: spec/react.spec.js (hunk 2 above)
Comment (chrisbolin) [reply spec/react.spec.js]: _references generate_, right?
I can switch it to _ reference generates_? is singular preferred?

Comment (ryan-roemer) [reply spec/react.spec.js]: I'm cool with whatever you want.

I try to have things read like sentences starting with `it` (being the test), so maybe `it warns on circular refs, but does not throw` -> `it('warns on circular refs, but does not throw', () => {`

But, totally up to you. 😄

**Pull request merged**